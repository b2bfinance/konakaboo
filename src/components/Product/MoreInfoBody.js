import Grid from '@material-ui/core/Grid';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Typography from '@material-ui/core/Typography';
import React from 'react';
import styled from 'styled-components';

export const Wrapper = styled(Grid)`
  padding: 16px;
`;

export const Disclaimer = styled.div`
  margin-bottom: 32px;
  background-color: ${props => props.theme.productBorder};
  padding: 16px;
`;

export const DetailRow = styled(TableRow)`
  height: 24px !important;
`;

export const DetailCell = styled(TableCell)`
  border: 0 !important;
`;

export default ({ detailed, disclaimer }) => (
  <Wrapper>
    {disclaimer && (
      <Disclaimer>
        <Typography>{disclaimer}</Typography>
      </Disclaimer>
    )}
    <Grid container spacing={4}>
      {detailed &&
        Object.keys(detailed).map(heading => (
          <Grid item xs={12} sm={6} lg={4} key={heading}>
            <Typography variant="h6">{heading}</Typography>
            <Table padding="none">
              <TableBody>
                {detailed[heading].map(detail =>
                  detail.label ? (
                    <DetailRow key={detail.label}>
                      <DetailCell variant="head">{detail.label}</DetailCell>
                      <DetailCell align="right">{detail.value}</DetailCell>
                    </DetailRow>
                  ) : (
                    <DetailRow key={detail.value}>
                      <DetailCell>
                        <Typography paragraph>{detail.value}</Typography>
                      </DetailCell>
                    </DetailRow>
                  )
                )}
              </TableBody>
            </Table>
          </Grid>
        ))}
    </Grid>
  </Wrapper>
);
