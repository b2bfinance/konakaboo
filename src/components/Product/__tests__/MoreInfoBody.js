import React from 'react';
import { render } from 'enzyme';
import MoreInfoBody, {
  Wrapper,
  Disclaimer,
  DetailRow,
  DetailCell
} from '../MoreInfoBody';

describe('Wrapper', () => {
  test('renders correctly', () => {
    expect(render(<Wrapper />)).toMatchSnapshot();
  });
});

describe('Disclaimer', () => {
  test('renders correctly', () => {
    expect(render(<Disclaimer />)).toMatchSnapshot();
  });
});

describe('DetailRow', () => {
  test('renders correctly', () => {
    expect(render(<DetailRow />)).toMatchSnapshot();
  });
});

describe('DetailCell', () => {
  test('renders correctly', () => {
    expect(render(<DetailCell />)).toMatchSnapshot();
  });
});

describe('MoreInfoBody', () => {
  test('renders correctly', () => {
    const product = stubData.products.item;
    expect(
      render(
        <MoreInfoBody
          detailed={product.detailed}
          disclaimer={product.disclaimer}
        />
      )
    ).toMatchSnapshot();
  });
});
