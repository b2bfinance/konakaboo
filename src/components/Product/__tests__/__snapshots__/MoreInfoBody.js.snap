// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailCell renders correctly 1`] = `
.c0 {
  border: 0 !important;
}

<td
  class="MuiTableCell-root-103 c0"
/>
`;

exports[`DetailRow renders correctly 1`] = `
.c0 {
  height: 24px !important;
}

<tr
  class="MuiTableRow-root-98 c0"
/>
`;

exports[`Disclaimer renders correctly 1`] = `
.c0 {
  margin-bottom: 32px;
  padding: 16px;
}

<div
  class="c0"
/>
`;

exports[`MoreInfoBody renders correctly 1`] = `
.c0 {
  padding: 16px;
}

.c1 {
  margin-bottom: 32px;
  padding: 16px;
}

.c2 {
  height: 24px !important;
}

.c3 {
  border: 0 !important;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <p
      class="MuiTypography-root-115 MuiTypography-body1-124"
    >
      Application for this product will impact your credit score.
    </p>
  </div>
  <div
    class="MuiGrid-container-1 MuiGrid-spacing-xs-32-26"
  >
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-lg-4-75"
    >
      <h6
        class="MuiTypography-root-115 MuiTypography-h6-132"
      >
        Highlight Points
      </h6>
      <table
        class="MuiTable-root-151"
      >
        <tbody
          class="MuiTableBody-root-152"
        >
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Max LTV
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              85%
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Initial Rate
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              1.49%
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Subsequent Rate
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              2.75%
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Fees
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              £1,495
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-lg-4-75"
    >
      <h6
        class="MuiTypography-root-115 MuiTypography-h6-132"
      >
        Technical Points
      </h6>
      <table
        class="MuiTable-root-151"
      >
        <tbody
          class="MuiTableBody-root-152"
        >
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Max LTV
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              85%
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Initial Rate
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              1.49%
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Subsequent Rate
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              2.75%
            </td>
          </tr>
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-head-104 MuiTableCell-paddingNone-110 c3"
            >
              Fees
            </td>
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-alignRight-113 MuiTableCell-paddingNone-110 c3"
            >
              £1,495
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <div
      class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-sm-6-49 MuiGrid-grid-lg-4-75"
    >
      <h6
        class="MuiTypography-root-115 MuiTypography-h6-132"
      >
        Point List
      </h6>
      <table
        class="MuiTable-root-151"
      >
        <tbody
          class="MuiTableBody-root-152"
        >
          <tr
            class="MuiTableRow-root-98 c2"
          >
            <td
              class="MuiTableCell-root-103 MuiTableCell-body-105 MuiTableCell-paddingNone-110 c3"
            >
              <p
                class="MuiTypography-root-115 MuiTypography-body1-124 MuiTypography-paragraph-143"
              >
                Lorem ipsum dolor sit amet, consectetur adipiscing elit.
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Wrapper renders correctly 1`] = `
.c0 {
  padding: 16px;
}

<div
  class="c0"
/>
`;
